IBX-4939: Doc changelog

DominikaK opened the pull request:
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/IBX-4939
| Versions      | all (?)
| Edition       | n/a

Add list of highlighted changes to documentation.

This would behave like product release notes, so merged to all branches and with absolute, not relative links

#### Checklist

- [x] Text renders correctly
- [x] Text has been checked with vale
- [x] Description metadata is up to date
- [x] Added link to this PR in relevant JIRA ticket or code PR

MagdalenaZuba commented:
We should also mention this:
- https://github.com/ezsystems/developer-documentation/pull/1881 
- https://github.com/ezsystems/developer-documentation/pull/1827 -  add info that we removed so many pages

DominikaK commented:
@MagdalenaZuba, I want to reserve this page for actual new stuff, not all major changes, especially ones less visible to readers. Plus, it's impossible to illustrate _removed_ pages, so such a note would be confusing (sort of "we've removed a thing and you can't check what thing that was, because we removed it")
